eventsourcing/src/test/java/org/axonframework/eventsourcing/eventstore/AnnotationBasedTagResolverTest.java

```diff
@@ -0,0 +1,701 @@
+                    @EventTag(key = "first"),
+                    @EventTag(key = "second")
+            })
+            // Adding repeatable annotation
+            @EventTag(key = "third")
+            private final String mixedField = "mixedValue";
+
+            // Using container annotation
+            @EventTags({
+                    @EventTag(key = "one"),
+                    @EventTag(key = "two")
+            })
+            // Adding repeatable annotation
+            @EventTag(key = "three")
+            public String getMixedValue() {
+                return "mixed";
+            }
+        }
+
+        @Test
+        void shouldHandleContainerAnnotation() {
+            // given
+            var payload = new ContainerAnnotationClass();
+            var event = anEventMessage(payload);
+
+            // when
+            var result = tagResolver.resolve(event);
+
+            // then
+            assertEquals(6, result.size());
+            // Tags from container annotation on 'id' field
+            assertTrue(result.contains(new Tag("id", "123")));
+            assertTrue(result.contains(new Tag("identifier", "123")));
+            // Tags from repeatable annotations on 'value' field
+            assertTrue(result.contains(new Tag("value", "456")));
+            assertTrue(result.contains(new Tag("amount", "456")));
+            // Tags from container annotation on getName method
+            assertTrue(result.contains(new Tag("methodName", "test")));
+            assertTrue(result.contains(new Tag("name", "test")));
+        }
+
+        @Test
+        void shouldHandleMixedContainerAndRepeatableAnnotations() {
+            // given
+            var payload = new MixedAnnotationClass();
+            var event = anEventMessage(payload);
+
+            // when
+            var result = tagResolver.resolve(event);
+
+            // then
+            assertEquals(6, result.size());
+            // Tags from field with mixed annotations
+            assertTrue(result.contains(new Tag("first", "mixedValue")));
+            assertTrue(result.contains(new Tag("second", "mixedValue")));
+            assertTrue(result.contains(new Tag("third", "mixedValue")));
+            // Tags from method with mixed annotations
+            assertTrue(result.contains(new Tag("one", "mixed")));
+            assertTrue(result.contains(new Tag("two", "mixed")));
+            assertTrue(result.contains(new Tag("three", "mixed")));
+        }
+
+        static class InvalidContainerClass {
+
+            @EventTags({}) // Empty container
+            private final String emptyContainer = "value";
+        }
+
+        @Test
+        void shouldHandleEmptyContainer() {
+            // given
+            var payload = new InvalidContainerClass();
+            var event = anEventMessage(payload);
+
+            // when
+            var result = tagResolver.resolve(event);
+
+            // then
+            assertTrue(result.isEmpty());
+        }
+    }
+
```

Comment:
Very complete here, awesome work! I did think of one other nested test class though, and that's with a custom annotation. There's a chance users don't want to use the `EventTag` or `EventTags` annotations, but rather have their own. Having that annotation meta-annotated with `EventTag` or `EventTags` would, ideally, still resolve the `Tags` for them.

Reply:
Ok, so this implementation doesn't work with meta annotations. Whereas `AnnotationUtils` handles meta-annotation, do not handle repeatable annotations... :/ So it's more complex. So I prefer to create follow-up issue for that.

Reply:
Yes, let's do that in a follow-up! Wasn't aware that the `AnnotationUtils` does not deal with repeatable at all. Would you mind drafting up that issue, @MateuszNaKodach? The priority can be 4, though. It's really just a nice to have at this stage.